ciStages.go

```diff
@@ -329,11 +373,31 @@ func makeDockerfile(config *helper.DockerBuildConfig, checkoutPath string) error
 
 func sendFailureNotification(failureMessage string, ciRequest *helper.CommonWorkflowRequest,
 	digest string, image string, ciMetrics helper.CIMetrics,
-	artifactUploaded bool, err error) (bool, error) {
+	artifactUploaded bool, err error) error {
 	e := helper.SendEvents(ciRequest, digest, image, ciMetrics, artifactUploaded, failureMessage, nil)
 	if e != nil {
 		log.Println(e)
-		return artifactUploaded, e
+		return e
 	}
-	return artifactUploaded, &helper.CiStageError{Err: err}
+	return &helper.CiStageError{Err: err}
+}
+
+func pushArtifact(ciCdRequest *helper.CiCdTriggerEvent, dest string, digest string, metrics *helper.CIMetrics, artifactUploaded bool) error {
+	imageRetryCountValue := ciCdRequest.CommonWorkflowRequest.ImageRetryCount
+	imageRetryIntervalValue := ciCdRequest.CommonWorkflowRequest.ImageRetryInterval
+	var err error
+	for i := 0; i < imageRetryCountValue+1; i++ {
+		if i != 0 {
+			time.Sleep(time.Duration(imageRetryIntervalValue) * time.Second)
+		}
+		err = helper.PushArtifact(dest)
+		if err == nil {
+			break
+		}
+	}
+	if err != nil {
```

Comment:
we are only handling the error found on last iteration, atleast log the errors found in previous iterations